include/zephyr/toolchain.h

```diff
@@ -38,15 +38,15 @@
  * and therefore this header is not meant to exist in-tree
  */
 #include <toolchain/other.h>
-#elif defined(__XCC__)
```

Comment:
If !defined(__llvm__) is not checked, xt-clang will #include <zephyr/toolchain/xcc.h>, because xt-clang also defines \_\_XCC\_\_ .